authority/options.go

```diff
@@ -42,3 +42,16 @@ func WithSSHGetHosts(fn func(cert *x509.Certificate) ([]sshutil.Host, error)) Op
 		a.sshGetHostsFunc = fn
 	}
 }
+
+// RootCAsGetter is an interface that gets a list of root certificates.
+type RootCAsGetter interface {
+	GetRootCertificates() []*x509.Certificate
+}
+
+// WithSSHAuthCheckHost sets a custom function to get the bastion for a
+// given user-host pair.
+func WithSSHAuthCheckHost(fn func(rootsGetter RootCAsGetter, tok string) error) Option {
```

Comment:
Change the function to something like `func(principal, token string, roots []*x509.Certificate)`